basic/src/test/java/org/jboss/hal/testsuite/test/runtime/transaction/TransactionsMetricsTestCase.java

```diff
@@ -0,0 +1,218 @@
+        NUMBER_OF_COMMITTED_TRANSACTIONS_LABEL = "Committed",
+        NUMBER_OF_COMMITTED_TRANSACTIONS = "number-of-committed-transactions",
+        NUMBER_OF_HEURISTICS_LABEL = "Heuristics",
+        NUMBER_OF_HEURISTICS = "number-of-heuristics",
+        NUMBER_OF_INFLIGHT_TRANSACTIONS_LABEL = "Inflight Transactions",
+        NUMBER_OF_INFLIGHT_TRANSACTIONS = "number-of-inflight-transactions",
+        NUMBER_OF_NESTED_TRANSACTIONS_LABEL = "Nested Transactions",
+        NUMBER_OF_NESTED_TRANSACTIONS = "number-of-nested-transactions",
+        NUMBER_OF_RESOURCE_ROLLBACKS_LABEL = "Resource Failures",
+        NUMBER_OF_RESOURCE_ROLLBACKS = "number-of-resource-rollbacks",
+        NUMBER_OF_SYSTEM_ROLLBACKS_LABEL = "System Failures",
+        NUMBER_OF_SYSTEM_ROLLBACKS = "number-of-system-rollbacks",
+        NUMBER_OF_TIMED_OUT_TRANSACTIONS_LABEL = "Timed Out",
+        NUMBER_OF_TIMED_OUT_TRANSACTIONS = "number-of-timed-out-transactions",
+        NUMBER_OF_TRANSACTIONS_LABEL = "Number of Transactions",
+        NUMBER_OF_TRANSACTIONS = "number-of-transactions",
+        TOTAL_FAILURES_LABEL = "Total Failures",
+        ARCHIVE_NAME = TransactionsMetricsTestCase.class.getSimpleName(),
+        STATISTICS_ENABLED = "statistics-enabled", TRANSACTIONS = "transactions";
+    private static final Address TXN_ADDRESS = Address.subsystem(TRANSACTIONS),
+            TXN_RUNTIME_ADDRESS = isDomain() ?
+                    Address.host(getDefaultHost()).and(SERVER, MONITORED_SERVER).and(SUBSYSTEM, TRANSACTIONS)
+                    : TXN_ADDRESS;
+    public static final int DELTA = 3;
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final Operations ops = new Operations(client);
+    private static final DeploymentsOperations deploymentOps = new DeploymentsOperations(client);
+    private static final Administration adminOps = new Administration(client);
+
+    private static boolean originallyStatisticEnabled;
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private TransactionsMetricsPage tmPage;
+
+    @BeforeClass
+    public static void deployTxnApplication() throws CommandFailedException, IOException, InterruptedException,
+            TimeoutException {
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar")
+                .addPackage("org.jboss.hal.testsuite.test.runtime.transaction.beans")
+                .addClass(TxnRollbackException.class)
+                .addAsManifestResource(new StringAsset("Dependencies: org.jboss.jts\n"), "MANIFEST.MF");
+        deploymentOps.deploy(jar);
+        originallyStatisticEnabled = ops.readAttribute(TXN_ADDRESS, STATISTICS_ENABLED).booleanValue();
+        if (!originallyStatisticEnabled) {
+            ops.writeAttribute(TXN_ADDRESS, STATISTICS_ENABLED, true);
+            adminOps.reloadIfRequired();
+        }
```

Comment:
The `originallyStatisticEnabled` should be rather remembering the `ModelNode` itself excluding default values in order to properly revert the changes after tests.

then you can always set it to true afterwards.